### PR TITLE
Add unique models to precommit PA tests

### PR DESCRIPTION
**Pull request by CuriousPanCake**
Add unique models to precommit PA tests

Change the list of models covering all the big real models.

Removed:
hf-tiny-model-private/tiny-random-GPT2LMHeadModel
hf-tiny-model-private/tiny-random-BartForCausalLM,xfail
hf-tiny-model-private/tiny-random-BigBirdForCausalLM,xfail
hf-tiny-model-private/tiny-random-BigBirdPegasusForCausalLM,xfail
hf-tiny-model-private/tiny-random-BlenderbotSmallForCausalLM,xfail
hf-tiny-model-private/tiny-random-ErnieForCausalLM,xfail
hf-tiny-model-private/tiny-random-GPTNeoXJapaneseForCausalLM,xfail
hf-tiny-model-private/tiny-random-MBartForCausalLM,xfail
hf-tiny-model-private/tiny-random-MvpForCausalLM,xfail
hf-tiny-model-private/tiny-random-PLBartForCausalLM,xfail
PrunaAI/hf-tiny-model-private-tiny-random-BloomForCausalLM-bnb-4bit-smashed,xfail
PrunaAI/hf-tiny-model-private-tiny-random-BloomForCausalLM-bnb-8bit-smashed,xfail
PrunaAI/hf-tiny-model-private-tiny-random-BloomForCausalLM-HQQ-2bit-smashed,xfail
hf-internal-testing/tiny-random-MegatronBertForCausalLM,xfail
hf-internal-testing/tiny-random-BlenderbotSmallForCausalLM,xfail

Added:
hf-internal-testing/Mixtral-tiny
hf-internal-testing/tiny-random-gpt2
katuni4ka/tiny-random-xverse
katuni4ka/tiny-random-baichuan2-13b
katuni4ka/tiny-random-qwen
katuni4ka/tiny-random-aquilachat
katuni4ka/tiny-random-aquila2
katuni4ka/tiny-random-qwen1.5-moe
katuni4ka/tiny-random-codegen2
katuni4ka/tiny-random-olmo-hf
katuni4ka/tiny-random-baichuan2
katuni4ka/tiny-random-jais
katuni4ka/tiny-random-internlm
katuni4ka/tiny-random-internlm2
katuni4ka/tiny-random-minicpm
fxmarty/tiny-random-GemmaForCausalLM
fxmarty/tiny-dummy-qwen2
fxmarty/really-tiny-falcon-testing
Xenova/tiny-random-Phi3ForCausalLM
facebook/opt-350m
katuni4ka/tiny-random-dbrx
katuni4ka/tiny-random-falcon-40b
katuni4ka/tiny-random-orion,xfail,No ScaledDotProductAttention operation observed in the graph 
katuni4ka/tiny-random-chatglm2,xfail,Model references undeclared parameters: beam_idx ()
katuni4ka/tiny-random-glm4,xfail,Model references undeclared parameters beam_idx () attention_mask ()

### Tickets:
 - CVS-145243

Signed-off-by: Andrii Staikov <andrii.staikov@intel.com>

### CODE REVIEW

File: tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit [hunk 1]
```diff
@@ -1,58 +1,46 @@
...
-hf-tiny-model-private/tiny-random-OPTForCausalLM,https://huggingface.co/hf-tiny-model-private/tiny-random-OPTForCausalLM
-hf-tiny-model-private/tiny-random-GPTJForCausalLM,https://huggingface.co/hf-tiny-model-private/tiny-random-GPTJForCausalLM
-hf-tiny-model-private/tiny-random-BloomForCausalLM,https://huggingface.co/hf-tiny-model-private/tiny-random-BloomForCausalLM
-hf-tiny-model-private/tiny-random-GPTNeoForCausalLM,https://huggingface.co/hf-tiny-model-private/tiny-random-GPTNeoForCausalLM
-hf-tiny-model-private/tiny-random-GPTNeoXForCausalLM,https://huggingface.co/hf-tiny-model-private/tiny-random-GPTNeoXForCausalLM
-hf-internal-testing/tiny-random-GPTNeoForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-GPTNeoForCausalLM
-hf-internal-testing/tiny-random-MptForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-MptForCausalLM
-hf-internal-testing/tiny-random-BloomForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-BloomForCausalLM
-hf-internal-testing/tiny-random-GPTJForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-GPTJForCausalLM
```
Comment (rkazants) [inline tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit:12]: can we leave them on the current lines? Please do not re-order - now it is hard to track what models are added and what are removed)

**Comment by mvafin** [comment]
Could you also in this PR fix here https://github.com/openvinotoolkit/openvino/blob/62a1f1b50c93268f96c325760442765115d01d74/.github/workflows/job_pytorch_models_tests.yml#L138

Currently these models run twice in validation, in `PyTorch Models Tests Not Timm or Torchvision` and in `PagedAttention Test`

File: tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit [hunk 2]
```diff
@@ -1,58 +1,46 @@
...
+hf-internal-testing/tiny-random-XGLMForCausalLM,https://huggingface.co/hf-tiny-model-private/tiny-random-XGLMForCausalLM,xfail,No ScaledDotProductAttention operation observed in the graph
```
Comment (itikhono) [inline tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit:1]: could you add a diff between old list and the new list: which model were deleted or changes, which ones were added

File: tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit [hunk 3]
```diff
@@ -1,58 +1,46 @@
...
-hf-internal-testing/tiny-random-MistralForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-OPTForCausalLM
+hf-internal-testing/tiny-random-GPTJForCausalLM,https://huggingface.co/trl-internal-testing/tiny-random-GPTJForCausalLM
+hf-internal-testing/tiny-random-GPTNeoForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-GPTNeoForCausalLM
 hf-internal-testing/tiny-random-GPTNeoXForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-GPTNeoXForCausalLM
-hf-internal-testing/tiny-random-LlamaForCausalLM,https://huggingface.co/trl-internal-testing/tiny-random-LlamaForCausalLM
-hf-internal-testing/tiny-random-StableLmForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-StableLmForCausalLM
-hf-internal-testing/tiny-random-PhiForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-PhiForCausalLM
+hf-internal-testing/tiny-random-MistralForCausalLM,https://huggingface.co/hf-internal-testing/tiny-random-MistralForCausalLM
+katuni4ka/tiny-random-orion,https://huggingface.co/katuni4ka/tiny-random-orion,xfail,looks like not supported
```
Comment (itikhono) [inline tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit:10]: could you combine models which failed with the same or similar issue, create jira ticket for them and add the ticket number as a comment to this file?

File: tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit (hunk 1 above)
Comment (CuriousPanCake) [reply tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit:12]: Added diff to the description

File: tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit (hunk 2 above)
Comment (CuriousPanCake) [reply tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit]: Added

File: tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit (hunk 3 above)
Comment (CuriousPanCake) [reply tests/model_hub_tests/pytorch/models/hf-tiny-random-models-precommit]: Fixed.

**Comment by CuriousPanCake** [comment]
> Could you also in this PR fix here
> 
> https://github.com/openvinotoolkit/openvino/blob/62a1f1b50c93268f96c325760442765115d01d74/.github/workflows/job_pytorch_models_tests.yml#L138
> 
> Currently these models run twice in validation, in `PyTorch Models Tests Not Timm or Torchvision` and in `PagedAttention Test`

Done

**Review by itikhono — APPROVED** [review]

**Comment by CuriousPanCake** [comment]
> > Could you also in this PR fix here
> > https://github.com/openvinotoolkit/openvino/blob/62a1f1b50c93268f96c325760442765115d01d74/.github/workflows/job_pytorch_models_tests.yml#L138
> > 
> > Currently these models run twice in validation, in `PyTorch Models Tests Not Timm or Torchvision` and in `PagedAttention Test`
> 
> Done

After discussing with @mvafin , we decided to leave it as a separate run.

**Pull request merged**